src/xcode/ENA/ENA/Source/Scenes/Checkin/CheckInDetail/TraceLocationDetailViewController.swift

```diff
@@ -0,0 +1,164 @@
+	@IBOutlet private weak var saveToDiaryLabel: ENALabel!
+	@IBOutlet private weak var automaticCheckOutLabel: ENALabel!
+	@IBOutlet private weak var pickerButton: ENAButton!
+	@IBOutlet private weak var pickerContainerView: UIView!
+	@IBOutlet private weak var pickerSeparator: UIView!
+	@IBOutlet private weak var datePickerView: UIDatePicker!
+	@IBOutlet private weak var additionalInfoView: UIView!
+	@IBOutlet private weak var additionalInfoLabel: ENALabel!
+	@IBOutlet private weak var pickerSwitch: ENASwitch!
+		
+	private let viewModel: TraceLocationDetailViewModel
+	private let dismiss: () -> Void
+	private var subscriptions = Set<AnyCancellable>()
+	private var isInitialSetup = true
+
+	private func setupView() {
+		view.backgroundColor = .enaColor(for: .background)
+		pickerButton.setTitleColor(.enaColor(for: .textPrimary1), for: .normal)
+		logoImageView.image = logoImageView.image?.withRenderingMode(.alwaysTemplate)
+		logoImageView.tintColor = .enaColor(for: .textContrast)
+		
+		addBorderAndColorToView(descriptionView, color: .enaColor(for: .hairline))
+		addBorderAndColorToView(bottomCardView, color: .enaColor(for: .hairline))
+		addBorderAndColorToView(additionalInfoView, color: .enaColor(for: .hairline))
+		
+		setupLabels()
+		setupPicker()
+		setupAdditionalInfoView()
+		
+		viewModel.$pickerButtonTitle
+			.sink { [weak self] hour in
+				if let hour = hour {
+					self?.pickerButton.setTitle(hour, for: .normal)
+				}
+			}
+			.store(in: &subscriptions)
+	}
+	
+	private func setupLabels() {
+		checkInForLabel.text = AppStrings.Checkins.Details.checkinFor
+		activityLabel.text = AppStrings.Checkins.Details.activity
+		saveToDiaryLabel.text = AppStrings.Checkins.Details.saveToDiary
+		automaticCheckOutLabel.text = AppStrings.Checkins.Details.automaticCheckout
+		descriptionLabel.text = viewModel.locationDescription
+		addressLabel.text = viewModel.locationAddress
+	}
+	
+	private func setupAdditionalInfoView() {
+		let status = viewModel.traceLocationStatus
+		switch status {
+		case .notStarted:
+			additionalInfoView.isHidden = false
+			additionalInfoLabel.text = String(
+				format: AppStrings.Checkins.Details.eventNotStartedYet,
+				viewModel.formattedString(for: .day) ?? "",
+				viewModel.formattedString(for: .hour) ?? ""
+			)
+		case .ended:
+			additionalInfoView.isHidden = false
+			additionalInfoLabel.text = AppStrings.Checkins.Details.eventEnded
+		case .inProgress:
+			additionalInfoView.isHidden = true
+		case .none:
+			additionalInfoView.isHidden = true
+		}
+	}
+	
+	private func setupPicker() {
+		datePickerView.locale = Locale(identifier: "de_DE")
+		datePickerView.datePickerMode = .countDownTimer
+		datePickerView.minuteInterval = 15
+		datePickerView.addTarget(self, action: #selector(didSelectDuration(datePicker:)), for: .valueChanged)
+	}
+	
+	@objc
+	private func didSelectDuration(datePicker: UIDatePicker) {
+		viewModel.pickerView(didSelectRow: datePicker.date.todaysMinutes)
```

Comment:
I think the clean way would be to use the `.countDownDuration` of the DatePicker

Reply:
we will get the values in seconds then we need to divide by 60, i did it anyway